### ISSUE TITLE
Download fails with "broken pipe" error

### ISSUE THREAD

**Issue by austinbutler** [ducaale/xh#135]
When downloading this particular file `xh` tends to fail.

```sh
❯ xh --download --output lid.176.bin https://dl.fbaipublicfiles.com/fasttext/supervised-models/lid.176.bin
HTTP/2.0 200 OK
accept-ranges: bytes
cf-cache-status: DYNAMIC
cf-ray: 647ae6814e4ae80d-LAX
cf-request-id: 09c0ae64d00000e80d05a72000000001
content-length: 131266198
content-type: application/octet-stream
date: Thu, 29 Apr 2021 19:22:40 GMT
etag: "be9a581239b678aafe80d1d4a46d86d8-16"
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
last-modified: Fri, 18 Jan 2019 15:01:14 GMT
server: cloudflare
set-cookie: __cfduid=dfd5b87e416587032a11e8c6f286918f91619724160; expires=Sat, 29-May-21 19:22:40 GMT; path=/; domain=.fbaipublicfiles.com; HttpOnly; SameSite=Lax
x-amz-id-2: umZpKluJeIllqq/c6zhTmEKE2/dYK0mJ7QBI6xtRpEMj1sko6eU1W2m970vyu07+/Ih7QglfQPA=
x-amz-request-id: C4P9T3MK474MWB9J
x-amz-server-side-encryption: AES256
x-amz-version-id: i0bvGpDpJiHbCgE2pdfmiT6e8rhcCJdD

Downloading 125.19MB to "lid.176.bin"
Error: request or response body error: error reading a body from connection: broken pipe

Caused by:
    0: error reading a body from connection: broken pipe
    1: broken pipe
```

On my macOS laptop it seems most consistently to fail. It also intermittently fails on two Arch Linux machines. Both `curl` and `httpie` consistently download the file without issue.

**Comment by blyxxyz** [comment]
I can reproduce this. I've opened an issue at hyper.

Using the OpenSSL backend fixes this, at least for this file on my connection. But we depend on rustls to read PEM files, so we'd have to compile with both rustls and OpenSSL, which adds 3 MB to the binary size.

**Comment by ducaale** [comment]
This should be now fixed in xh v0.10.0. Feel free to reopen the issue if the issue still exits

**Comment by austinbutler** [comment]
Just upgraded and can confirm it's fixed.
